Restore wiimote netplay

mimimi085181 opened the pull request:
In its broken state.

Reverts prs https://github.com/dolphin-emu/dolphin/pull/3802, https://github.com/dolphin-emu/dolphin/pull/3691 and https://github.com/dolphin-emu/dolphin/pull/3660

The broken state will be useful when trying to fix wiimote netplay.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dolphin-emu/dolphin/3949)
<!-- Reviewable:end -->

Jofzar commented:
The broken state also breaks testing for netplay, there are 3 netplay changes waiting to be merged, could it be possible to merge and test those before this is broken again?

mimimi085181 commented:
Are you sure it breaks netplay? You should just have to disable all wimotes manually.

Jofzar commented:
I would have to retest but at one point netplay was straight breaking due to one of these, also #3691 changes the text of the netplay window, which should probably be noted for consistency when these are reverted.

mimimi085181 commented:
I need this pr merged, before i can really contine to work on it. I have one follow up pr ready, which should fix starting a 2nd netplay session. The code static u8 previousSize[4] = {4, 4, 4, 4}; only sets the size for the 7 dummy inputs to 4 for the 1st session, but on the 2nd start, it has the size where the previous session ended.

@Jofzar: Please check how i changed the texts. I reverted it, and added a line "Manually set the extensions for each Wiimote".

And also, netplay with gamecube controllers should work fine with this merged, if you manually set all wiimotes to none on all clients. @JMC47: Can you confirm this?

JMC47 commented:
@lioncash can you review this?  I don't know who else to bother, but this is becoming important.

Jofzar commented:
Seems to be fine, would prefer to have a second opinion, my game library is smaller then jmc's so its probably best for his opinion.

As for the wording, I think there could be a better non technical way of saying that.

Maybe something like this?

"If you are using Wii Motes, they will have to be manually set"

mimimi085181 commented:
@Jofzar: One big problem with the current code is that there's no difference between the setup remote and local wiimotes. If player 2 uses a nunchuck for example, the 2nd controller needs to have the nunchuck extension set on both clients. The problem is that the input is taken from the 1st slot from player 2, so that has to be set to nunchuck as well.

So maybe something like: "All players need to use the same Wiimote extensions?"?

Jofzar commented:
Add on the end "to avoid desyncing"

mathieui commented:
Reviewed 2 of 5 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3949)*
<!-- Sent from Reviewable.io -->

mathieui commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3949)*
<!-- Sent from Reviewable.io -->

mimimi085181 commented:
For the record, before the 3 prs, Dolphin properly disabled all wiimotes when there's no mapped wiimote for netplay. I got confused because somewhen it was necessary to disable wiimote manually, which is not the case after reverting all 3 prs.
